pkg/miscellaneous/miscellaneous_test.go

```diff
@@ -586,28 +587,164 @@ var _ = Describe("Miscellaneous Tests", func() {
 		})
 	})
 
-	Describe("parse peer urls config", func() {
-		var (
-			initialAdPeerURL string
-			podName          string
+	Describe("Get Advertise URLs", func() {
+		const (
+			configFile = "/tmp/etcd-config.yaml"
+			podName    = "test-pod"
 		)
-
-		BeforeEach(func() {
-			podName = "etcd-test-pod-0"
+		type testCase struct {
+			name     string
+			field    string
+			function func(string) ([]string, error)
+		}
+		testCases := []testCase{
```

Comment:
You can use [ginkgo Table Specs  ](https://onsi.github.io/ginkgo/#table-specs) for this case.

Reply:
I've now used ginkgo table driven tests, pls check.